Add theme checks, `Minimal diff` tab and  `Set As Diff Base` button

six-two opened the pull request:
Hi,

thank you for the extension. I found it very useful and I improved a couple things:

1. Added a check whether Burpsuite is in dark or light theme. For the light theme I chose different colors that make the text better to read.
2. Added a second tab called `Minimal diff` that displays only changes and on which line in the original request they occurred.
3. Added a `Set As Diff Base` button to the top of the diff tabs, that does the same thing as the context menu. But it is faster to use, especially when playing around with many requests in a repeater tab.

Below is a screenshot that shows all three changes in action:
![2025-03-14_20-19](https://github.com/user-attachments/assets/05995ba4-a279-49ab-a31a-5cb0cd52552e)

The code is not well tested yet, so it might contain some bugs.

Best regards,
six-two

six-two commented:
I noticed, that I could not see which text I have selected in the highlighted sections. To fix it, I added some transparency (alpha) to the added/deleted highlighting. The exact alpha and base colors probably need some tweaking, but at least the selected text is slightly differently colored now.

b-faller commented on src/main/java/burp/CustomHighlight.java:
```diff
@@ -53,6 +61,17 @@ class CustomHttpResponseEditor implements ExtensionProvidedHttpResponseEditor, P
+        setBaselineButton.addActionListener(e -> {
+            HttpResponse response = getResponse();
+            Optional<HttpResponse> responseAsOptional = (response != null) ? Optional.of(response) : Optional.empty();
+            BaseResponse.setBaseResponse(responseAsOptional);
```
You should be able to do:

```suggestion
            Optional<HttpResponse> response = Optional.ofNullable(getResponse());
            BaseResponse.setBaseResponse(response);
```

six-two replied:
Implemented

b-faller commented on src/main/java/burp/CustomHighlight.java:
```diff
@@ -134,30 +165,33 @@ public String generateDiff(String baseText, String newText, List<CustomHighlight
                     delta.getTarget().getLines().forEach(line -> result.append(line).append("\n"));
                     endLine = startLine + delta.getTarget().getLines().size();
-                    highlighters.add(new CustomHighlight(startLine, endLine, new Color(55, 69, 102)));
+                    highlighters.add(new CustomHighlight(startLine, endLine, insertColor));
```
I would keep different colors for add/remove vs. changes to provide a bit more visual feedback.

b-faller reviewed (commented):
Hi, thank you for the great contribution, nice improvements!

I left some small notes. In the long run I think it would make sense to have only one tab which can be switched between the views in the same row as the new button.

six-two commented:
@b-faller Thank you for the suggestions:

1. I changed the code to use one tab with a drop down to select the diff type. 
2. Regarding the color comment: I added `getChangeNewColor` and `getChangeOldColor` to `src/main/java/burp/DiffHelper.java`, which can be overwritten to have different colors for changed text. As I am not sure what coloring scheme you would like best, I have not given them different colors yet. Feel free to modify any of my color settings, they are probably not the best.

six-two commented:
I added a new feature: the Diff tab is also available for the request. This can for example be useful to see which parameters (ids, CSFR tokens, etc) change when you perform the same action via the application twice.

b-faller reviewed (approved):
Those are some great changes, thank you again!
